### ISSUE TITLE
[BUG] vcsim does not set VAppProductInfo on VirtualMachine

### ISSUE THREAD

**Issue by jcpowermac** [vmware/govmomi#3111]
**Describe the bug**

vcsim does not set VAppProductInfo on imported or created VirtualMachine

**To Reproduce**

```bash
wget https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.12/latest/rhcos-4.12.10-x86_64-vmware.x86_64.ova
govc import.ova -pool /DC0/host/DC0_C0 -dc DC0 rhcos-4.12.10-x86_64-vmware.x86_64.ova 
govc vm.info -json '/DC0/vm/OpenShift 4' | jq
```

```golang

	config := types.VirtualMachineConfigSpec{
		Name:    name,
		GuestId: string(types.VirtualMachineGuestOsIdentifierOtherGuest),
		Files: &types.VirtualMachineFileInfo{
			VmPathName: "[LocalDS_0]",
		},
		VAppConfig: &types.VAppConfigSpec{
			VmConfigSpec: types.VmConfigSpec{
				Product: []types.VAppProductSpec{{
					ArrayUpdateSpec: types.ArrayUpdateSpec{
						Operation: types.ArrayUpdateOperationAdd,
					},
					Info: &types.VAppProductInfo{
						Key:     int32(0),
						Name:    "rhcos OpenShift 4",
						Vendor:  "rhcos",
						Version: "foo",
					},
				}},
			},
		},
	}

	var devices object.VirtualDeviceList
	vmnet := esx.EthernetCard.Backing
	nic := esx.EthernetCard
	nic.Backing = vmnet

	scsi, _ := devices.CreateSCSIController("pvscsi")
	ide, _ := devices.CreateIDEController()
	cdrom, _ := devices.CreateCdrom(ide.(*types.VirtualIDEController))
	disk := devices.CreateDisk(scsi.(types.BaseVirtualController), datastores[0].Reference(),
		config.Files.VmPathName+" "+path.Join(name, "disk1.vmdk"))
	disk.CapacityInKB = int64(units.GB*10) / units.KB
	disk.StorageIOAllocation = &types.StorageIOAllocationInfo{Limit: types.NewInt64(-1)}

	devices = append(devices, scsi, cdrom, disk, &nic)

	config.DeviceChange, _ = devices.ConfigSpec(types.VirtualDeviceConfigSpecOperationAdd)


	task, err := vmFolder.CreateVM(ctx, config, resourcePools[0], hosts[0])
	if err != nil {
		log.Fatal(err)
	}

	info, err := task.WaitForResult(ctx, nil)
	if err != nil {
		log.Fatal(err)
	}

	rhcos := object.NewVirtualMachine(c.Client, info.Result.(types.ManagedObjectReference))
	spew.Dump(rhcos)
	vms, err := finder.VirtualMachineList(ctx, "/...")

	if err != nil {
		log.Fatal(err)
	}

	for _, v := range vms {
		fmt.Println(v.Name())

		if strings.Contains(v.Name(), "rhcos") {
			var vmMo mo.VirtualMachine

			err = v.Properties(ctx, v.Reference(), []string{"config"}, &vmMo)

			if err != nil {
				log.Fatal(err)
			}
			if vmMo.Config != nil {
				if vmMo.Config.VAppConfig != nil {
					spew.Dump(vmMo.Config.VAppConfig)

				}
			}
		}
	}
```

**Expected behavior**
Configuration spec of virtual machine set should be applied to VirtualMachine.

**Affected version**
0.30.4, main

**Screenshots/Debug Output**
If applicable, add screenshots or debug output to help explain your problem.

**Additional context**

**Comment by dougm** [comment]
@jcpowermac sorry for the delay, fixed by #3252 if you want to give it a try

**Comment by jcpowermac** [comment]
> @jcpowermac sorry for the delay, fixed by #3252 if you want to give it a try

No worries @dougm thanks for fixing it! Though now I have to remember what I was up to ;-)
